core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -345,38 +345,44 @@ public static String deletePropertiesStatement(Set<String> properties) {
 
   static boolean namespaceExists(
       String catalogName, JdbcClientPool connections, Namespace namespace) {
-    if (exists(
+    if (namespaceExists(
         connections,
         JdbcUtil.GET_NAMESPACE_SQL,
         catalogName,
-        JdbcUtil.namespaceToString(namespace) + "%")) {
+        JdbcUtil.namespaceToString(namespace))) {
       return true;
     }
 
-    if (exists(
+    if (namespaceExists(
         connections,
         JdbcUtil.GET_NAMESPACE_PROPERTIES_SQL,
         catalogName,
-        JdbcUtil.namespaceToString(namespace) + "%")) {
+        JdbcUtil.namespaceToString(namespace))) {
       return true;
     }
 
     return false;
   }
 
   @SuppressWarnings("checkstyle:NestedTryDepth")
-  private static boolean exists(JdbcClientPool connections, String sql, String... args) {
+  private static boolean namespaceExists(
+      JdbcClientPool connections, String sql, String catalogName, String namespace) {
```

Comment:
I think you've misinterpreted the purpose of this method. This checks whether any entity exists that matches the query -- whether there is at least one record returned. I don't see any reason to change the fact that this method is generic.

Reply:
rolled this change back